ek9000App/src/scripts/ci_check.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+import json
+import argparse
+import jsonschema
+import os
+
+# Reference: https://json-schema.org/specification.html
+SCHEMA = {
+    'type': 'object',
+    'properties': {
+        'terminals': {
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'required': ['name', 'type', 'inputs', 'outputs', 'pdo_in_size', 'pdo_out_size'],
+                'properties': {
+                    'name': {
+                        'type': 'string'
+                    },
+                    'type': {
+                        'type': ['string', 'array'],
+                        'if': {
+                            'type': 'string'
+                        },
+                        'then': {
+                            'enum': ['AnalogIn', 'AnalogOut', 'DigIn', 'DigOut', 'PositionMeasurement', 'DigInMulti', 'DigOutMulti']
+                        },
+                        'else': {
+                            'items': {
+                                'type': 'string',
+                                'enum': ['AnalogIn', 'AnalogOut', 'DigIn', 'DigOut', 'PositionMeasurement', 'DigInMulti', 'DigOutMulti']
+                            },
+                        }
+                    },
+                    'inputs': {
+                        'type': 'number',
+                        'minimum': 0
+                    },
+                    'outputs': {
+                        'type': 'number',
+                        'minimum': 0
+                    },
+                    'pdo_in_size': {
+                        'type': 'number',
+                        'minimum': 0
+                    },
+                    'pdo_out_size': {
+                        'type': 'number',
+                        'minimum': 0
+                    },
+                    'dtyp': {
+                        'type': 'string'
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+def get_scripts_dir() -> str:
+    return os.path.dirname(__file__)
+
+
+def get_db_dir() -> str:
+    return f'{get_scripts_dir()}/../../Db'
+
+
+def check_json(file: dict) -> bool:
+    print('Validating terminals.json...', end='')
+    jsonschema.validate(file, SCHEMA)
```

Comment:
Just double-checking: `jsonschema.validate()` will raise on error: https://python-jsonschema.readthedocs.io/en/stable/validate/ so this is fine

Reply:
Yup